Game/Source/GDKShooter/Private/UI/TouchControls.cpp

```diff
@@ -0,0 +1,352 @@
+	{
+		ControllerEvents->DeathDetailsEvent.AddDynamic(this, &UTouchControls::OnDeath);
+	}
+}
+
+void UTouchControls::OnDeath(const FString& VictimName, int32 VictimId)
+{
+	SetVisibility(ESlateVisibility::Hidden);
+}
+
+void UTouchControls::BindControls()
+{
+	if (!bControlsBound)
+	{
+		JumpButton->OnMouseButtonDownEvent.BindDynamic(this, &UTouchControls::HandleJumpPressed);
+		JumpButton->OnMouseButtonUpEvent.BindDynamic(this, &UTouchControls::HandleJumpReleased);
+
+		CrouchSlideButton->OnMouseButtonDownEvent.BindDynamic(this, &UTouchControls::HandleCrouchPressed);
+		CrouchSlideButton->OnMouseButtonUpEvent.BindDynamic(this, &UTouchControls::HandleCrouchReleased);
+
+		RightShootButton->OnMouseButtonDownEvent.BindDynamic(this, &UTouchControls::HandleTriggerPressed);
+		RightShootButton->OnMouseButtonUpEvent.BindDynamic(this, &UTouchControls::HandleTriggerReleased);
+
+		SprintButton->OnMouseButtonDownEvent.BindDynamic(this, &UTouchControls::HandleSprintPressed);
+		SprintButton->OnMouseButtonUpEvent.BindDynamic(this, &UTouchControls::HandleSprintReleased);
+
+		SiteScopeButton->OnMouseButtonDownEvent.BindDynamic(this, &UTouchControls::HandleScopePressed);
+		SiteScopeButton->OnMouseButtonUpEvent.BindDynamic(this, &UTouchControls::HandleScopeReleased);
+
+		bControlsBound = true;
+	}
+}
+
+FReply UTouchControls::NativeOnTouchStarted(const FGeometry& InGeometry, const FPointerEvent& InGestureEvent)
+{
+	const uint32 TouchIndex = InGestureEvent.GetPointerIndex();
+	const FVector2D& ScreenSize = InGeometry.GetLocalSize();
+	const FVector2D LocalPosition = InGeometry.AbsoluteToLocal(InGestureEvent.GetScreenSpacePosition());
+	UE_LOG(LogTouchControls, Verbose, TEXT("NativeOnTouchStarted, Pointer index: %d"), TouchIndex);
+
+	if (InLeftControllerResponseArea(LocalPosition, ScreenSize))
+	{
+		LeftControllerInfo.LeftTouchStartPosition = LocalPosition;
+		LeftControllerInfo.LeftControllerActive = true;
+		LeftControllerInfo.TouchIndex = TouchIndex;
+		return FReply::Handled();
+	}
+	else
+	{
+		if (!CameraTouch.active)
+		{
+			CameraTouch.LastPos = LocalPosition;
+			CameraTouch.FingerIndex = TouchIndex;
+			CameraTouch.active = true;
+			return FReply::Handled();
+		}
+	}
+	return FReply::Unhandled();
+}
+
+FReply UTouchControls::NativeOnTouchMoved(const FGeometry& InGeometry, const FPointerEvent& InGestureEvent)
+{
+	const uint32 TouchIndex = InGestureEvent.GetPointerIndex();
+	const FVector2D LocalPosition = InGeometry.AbsoluteToLocal(InGestureEvent.GetScreenSpacePosition());
+	if (LeftControllerInfo.LeftControllerActive && LeftControllerInfo.TouchIndex == TouchIndex)
+	{
+		HandleTouchMoveOnLeftController(LocalPosition);
+		return FReply::Handled();
+	}
+	else if (CameraTouch.active && CameraTouch.FingerIndex == TouchIndex)
+	{
+		HandleTouchMoveOnRightController(LocalPosition);
+		return FReply::Handled();
+	}
+	else
+	{
+		return FReply::Unhandled();
+	}
```

Comment:
```suggestion
	return FReply::Unhandled();
```
nit: I guess there is no need for an else? Either it got handled in the previous if statements or it fails.